go/runtime/scheduling/simple/txpool/priorityqueue/priority_queue.go

```diff
@@ -138,15 +156,35 @@ func (q *priorityQueue) GetBatch(force bool) []*transaction.CheckedTransaction {
 	// Remove transactions discovered to be too big to even fit the batch.
 	// This can happen if weight limits changed after the transaction was
 	// already set to be scheduled.
-	for _, item := range toRemove {
+	q.removeTxsLocked(toRemove)
+
+	return batch
+}
+
+func (q *priorityQueue) removeTxsLocked(items []*item) {
+	for _, item := range items {
 		delete(q.transactions, item.tx.Hash())
 		q.priorityIndex.Delete(item)
 		for k, v := range item.tx.Weights() {
 			q.poolWeights[k] -= v
 		}
 	}
 
-	return batch
+	// Update lowest priority.
+	if len(items) > 0 {
+		if lpi := q.priorityIndex.Max(); lpi != nil {
```

Comment:
Should we invert the `item.Less` function so that we can use `Min()` here, as that would make more sense (I think only `Max`->`Min` and `Ascend` -> `Descend`) will need update? 

I don't recall if there's any special reason why `Less` is currently inverted, I don't think it should impact the overall performance changing it.

Reply:
Yeah makes sense, I also found this confusing.

Reply:
Updated.